src/lib/datepicker/calendar.spec.ts

```diff
@@ -231,6 +246,347 @@ describe('MdCalendar', () => {
 
       expect(testComponent.selected).toEqual(new SimpleDate(2017, 0, 2));
     });
+
+    describe('a11y', () => {
+      let calendarBodyEl: HTMLElement;
+
+      beforeEach(() => {
+        calendarBodyEl = calendarElement.querySelector('.mat-calendar-body') as HTMLElement;
+        expect(calendarBodyEl).not.toBeNull();
+
+        dispatchFakeEvent(calendarBodyEl, 'focus');
+        fixture.detectChanges();
+      });
+
+      it('should not allow selection of disabled date in month view', () => {
+        expect(calendarInstance._monthView).toBe(true);
+        expect(calendarInstance._activeDate).toEqual(new SimpleDate(2017, 0, 1));
```

Comment:
Yeah this was more of a practical decision, since someone could theoretically create a calendar with no maxDate but a date filter that disables all dates > 1/1/17. In this case if the user tried to arrow past 1/1/17 everything would just lock up as it searched for the next enabled date. I put a TODO and I'm curious to see how AngularJS Material handled this